libraries/apollo-tooling/src/main/kotlin/com/apollographql/apollo3/tooling/RegisterOperations.kt

```diff
@@ -255,30 +254,34 @@ object RegisterOperations {
 
     val response = runBlocking { call.execute() }
     val data = response.data
-    if (data == null) {
-      when (val e = response.exception!!) {
-        is ApolloHttpException -> {
-          val body = e.body?.use { it.readUtf8() } ?: ""
-          throw Exception("Cannot push operations: (code: ${e.statusCode})\n$body", e)
-        }
-
-        is ApolloGraphQLException -> {
-          throw Exception("Cannot push operations: ${e.errors.joinToString { it.message }}")
-        }
-
-        else -> {
-          throw Exception("Cannot push operations: ${e.message}", e)
-        }
+    if (data != null) {
+      val service = data.service
+      if (service == null) {
+        throw Exception("Cannot push operations: cannot find service '$graphID': ${response.errors?.joinToString { it.message }}")
```

Comment:
This is somewhat inconsistent. If we expect an error on null, we should make `service` `@semanticNonNull`. Do you remember what's happening in those cases?

Reply:
Ah, it's the deprecated one, apologies. Let's keep this one as is for now but there is also the issue for `FieldInsights`, I'll comment there.

Reply:
Added some deprecation and comments [there](https://github.com/apollographql/apollo-kotlin/commit/79e68b16b33268cb854b66dc55627ac3fbcb7ad3)